src/config.ts

```diff
@@ -218,6 +218,12 @@ export interface Config {
 
   /** Selection Config */
   selection?: SelectionConfig;
+
+  // Support arbitrary key for role config
+  // Note: Technically, the type for role config should be MarkConfig.
+  // However, Typescript requires that the index type must be compatible with all other properties.
+  // Thus, we have to use any here.
```

Comment:
What does "Typescript requires that the index type must be compatible with all other properties" mean?